sources/core.ts

```diff
@@ -379,12 +383,27 @@ export function selectBestState(input: Array<string>, states: Array<RunState>) {
   if (terminalStates.length === 0)
     throw new Error();
 
+  const requiredOptionsSetStates = terminalStates.filter(state =>
+    state.requiredOptions.every(names =>
+      names.some(name =>
+        state.options.find(opt => opt.name === name)
+      )
+    )
+  );
```

Comment:
For future reference: I considered suggesting to make `state.requiredOptions` a set, so that we could simply remove the option each time we pushed an option rather than these nested iterations (non-empty sets would then not match).

However, after more consideration, it would actually make the code more complex since we'd have to add those `delete` calls in most of the reducers. Better to make a classic iteration instead, especially since it shouldn't have any cost on most commands.